docs/rfcs/rfc-029-unwind-create-auth.md

```diff
@@ -0,0 +1,1401 @@
+    }
+    RETURN create_this1
+}
+CALL {
+    WITH create_this1
+    MATCH (create_this1_moderators:`User`)-[create_this0:MODERATES_POST]->(create_this1)
+    WITH create_this1_moderators { .name } AS create_this1_moderators
+    RETURN collect(create_this1_moderators) AS create_this1_moderators
+}
+RETURN collect(create_this1 { .title, .content, moderators: create_this1_moderators }) AS data
+```
+
+### Entity-type Bind - across relationship
+
+Bind could be used across relationship, a valid use case it may be "ensure that users only create Posts related to themselves".
+
+This could be achieved in the unwind-create context by adding the following validation block to the Entity where the `@auth` is definded:
+
+```cypher
+    WITH create_this1
+    CALL {
+            WITH create_this1
+            CALL apoc.util.validate(
+                NOT ((exists((create_this1)<-[:HAS_POST]-(:`User`)) 
+                    AND 
+            all(auth_this0 IN [(create_this1)<-[:HAS_POST]-(auth_this0:`User`) | auth_this0]
+                 WHERE (auth_this0.id IS NOT NULL AND auth_this0.id = $this0auth_param0))))
+                 , "@neo4j/graphql/FORBIDDEN", [0])
+    }
+```
+For instance:
+
+```cypher
+UNWIND [
+    {
+      "title": "A wonderful title!",
+      "content": "A wonderful post!",
+      "creator": {
+        "create": {
+          "node": {
+            "id": "1234567890",
+            "name": "Simone"
+          }
+        }
+      }
+    }
+  ] AS create_var2
+CALL {
+    WITH create_var2
+    CREATE (create_this1:`Post`)
+    SET
+        create_this1.title = create_var2.title,
+        create_this1.content = create_var2.content
+    WITH create_this1, create_var2
+    CALL {
+        WITH create_this1, create_var2
+        UNWIND create_var2.creator.create AS create_var3
+        WITH create_var3.node AS create_var4, create_var3.edge AS create_var5, create_this1
+        CREATE (create_this6:`User`)
+        SET
+            create_this6.id = create_var4.id,
+            create_this6.name = create_var4.name
+        MERGE (create_this6)-[create_this7:HAS_POST]->(create_this1)
+        
+        RETURN collect(NULL)
+    }
+    WITH create_this1
+    CALL {
+        WITH create_this1
+        MATCH (create_this1)<-[create_this1_creator_User_unique:HAS_POST]-(:User)
+        WITH count(create_this1_creator_User_unique) as c
+        CALL apoc.util.validate(NOT (c = 1), '@neo4j/graphql/RELATIONSHIP-REQUIREDPost.creator required', [0])
+        RETURN c AS create_this1_creator_User_unique_ignored
+    }
+    WITH create_this1
+    CALL {
+            WITH create_this1
+            CALL apoc.util.validate(
+                NOT ((exists((create_this1)<-[:HAS_POST]-(:`User`)) 
+                    AND 
+            all(auth_this0 IN [(create_this1)<-[:HAS_POST]-(auth_this0:`User`) | auth_this0]
+                 WHERE (auth_this0.id IS NOT NULL AND auth_this0.id = $this0auth_param0))))
+                 , "@neo4j/graphql/FORBIDDEN", [0])
```

Comment:
This whole Cypher needs a whole lot of modernising, but unsure if that's outside of the scope of this work:

```suggestion
            WHERE apoc.util.validatePredicate(
                NOT EXISTS { (create_this1)<-[:HAS_POST]-(:`User`) }
                    AND 
                NOT EXISTS { (create_this1)<-[:HAS_POST]-(auth_this0:`User`) 
                 WHERE NOT auth_this0.id IS NOT NULL AND auth_this0.id = $this0auth_param0 }
                 , "@neo4j/graphql/FORBIDDEN", [0])
```